### PR TITLE
Stops vending machines from attacking SSD people and spamming the logs

### PR DESCRIPTION
**Pull request by Ty-Omaha**
**What does this PR do:**
Stops vending machines creating attack logs by throwing stuff at SSD players.

:cl:
tweak: Vending Machines will no longer throw items at SSD people.
/ :cl:

### CODE REVIEW

File: code/game/machinery/vending.dm [hunk 1]
```diff
@@ -708,7 +708,7 @@
 /obj/machinery/vending/proc/throw_item()
 	var/obj/throw_item = null
 	var/mob/living/target = locate() in view(7,src)
-	if(!target)
+	if(!target || !target.client)
```
Comment (Citinited) [inline code/game/machinery/vending.dm]: Shouldn't this try to find another valid target if there is one instead of just `return`ing?

Comment (farie82) [reply code/game/machinery/vending.dm]: throw the ssd infront of rogue machines to make them stop firing!

Comment (Ty-Omaha) [reply code/game/machinery/vending.dm]: @farie82 it would only stop firing if it was already targeting the SSD player

Comment (Ty-Omaha) [reply code/game/machinery/vending.dm]: @Citinited also that opens the possibility of endless loops

Comment (Fox-McCloud) [reply code/game/machinery/vending.dm]: This means monkies and mobs won't be hit by this anymore.

I don't like that.

Is this *really* that much of a problem?

Comment (Ty-Omaha) [reply code/game/machinery/vending.dm]: yeah it spams the logs a whole lot. I'll just switch to a player_logged check then

File: code/game/machinery/vending.dm [hunk 2]
```diff
@@ -710,7 +710,11 @@
 	var/mob/living/target = locate() in view(7,src)
```
Comment (Fox-McCloud) [inline code/game/machinery/vending.dm]: would probably be better to build a temporay list of all living mobs in `view(7, src)`, iterate over it, skip SSDs, then pick, from that list, which one to fire at.

I suspect `locate` has biases in it, anyway.

File: code/game/machinery/vending.dm [hunk 3]
```diff
@@ -710,7 +710,11 @@
 	var/mob/living/target = locate() in view(7,src)
 	if(!target)
 		return 0
-
+	if(istype(target, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = target
+		if(H.player_logged)
+			return 0
```
Comment (Fox-McCloud) [inline code/game/machinery/vending.dm]: we have an SSD helper proc now.

Comment (Ty-Omaha) [reply code/game/machinery/vending.dm]: what is the SSD helper proc?

**Comment by variableundefined** [comment]
@Ty-Omaha IsLivingSSD(mob/M)

**Comment by Ty-Omaha** [comment]
don't have the time to make this loop through

**Pull request closed without merging**